### PR TITLE
Adds support for disabling TTS Autoplay

### LINKED ISSUE #161: Detect Autoplay Audio Restriction on iPhones and Display Play Button
On iPhones, there is a known restriction that prevents audio from autoplaying without user interaction. This issue results in our audio content only playing when manually triggered by the user, rather than auto-playing as it does on other devices.

### Details:
- **Platform Affected:** iOS (iPhones)
- **Behavior Observed:** Audio does not autoplay. Requires a manual play button press for each playback instance.
- **Reference:** Apple's [Using HTML5 Audio and Video guide](https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/Device-SpecificConsiderations/Device-SpecificConsiderations.html) outlines this restriction.

### Proposed Solution:
Implement a feature to detect if the user is accessing our service from an iPhone. If so, rather than attempting to autoplay audio (which will fail due to Apple's restrictions), we should display a play button for user interaction. This approach is in line with user experience on platforms like YouTube, where user interaction is required to start audio playback.

### Expected Outcome:
- For iPhone users: Display a play button to initiate audio playback.
- For users on other devices: Audio should autoplay as currently implemented with the play/pause button.

### Additional Notes:
- It's crucial that this solution is iPhone-specific and does not impact the user experience on other devices.

### PR DESCRIPTION
**Pull request by rithviknishad**
Also fixes #161

### CODE REVIEW

File: src/utils/misc.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/**
+ * `true` if device is an iOS device, else `false`
+ */
+export const isIOS = () => /iPad|iPhone|iPod/.test(navigator.userAgent);
```
Comment (skks1212) [inline src/utils/misc.ts:4]: @rithviknishad we don't face the problems with macs right?

Comment (Ashesh3) [reply src/utils/misc.ts:4]: @skks1212 No issues with MacOS.

**Comment by skks1212** [comment]
@rithviknishad 
Build failing

```
<html>
<body>
<!--StartFragment-->
ReferenceError: navigator is not defined
--
20:29:03.978 | at a (/vercel/path0/build/server/chunks/293.js:1:12801)
20:29:03.978 | at v (/vercel/path0/build/server/chunks/293.js:1:5186)
20:29:03.978 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:45321)
20:29:03.978 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.978 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.978 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:56390)
20:29:03.978 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.978 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.978 | at nO (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64035)
20:29:03.978 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:62224)
20:29:03.979 |  
20:29:03.979 | Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
20:29:03.979 | ReferenceError: navigator is not defined
20:29:03.979 | at a (/vercel/path0/build/server/chunks/293.js:1:12801)
20:29:03.979 | at v (/vercel/path0/build/server/chunks/293.js:1:5186)
20:29:03.979 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:45321)
20:29:03.979 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.979 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.979 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:56390)
20:29:03.979 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.979 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.979 | at nO (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64035)
20:29:03.979 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:62224)
20:29:03.979 | ReferenceError: navigator is not defined
20:29:03.979 | at a (/vercel/path0/build/server/chunks/293.js:1:12801)
20:29:03.979 | at v (/vercel/path0/build/server/chunks/293.js:1:5186)
20:29:03.979 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:45321)
20:29:03.979 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.980 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.980 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:56390)
20:29:03.980 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.980 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.980 | at nO (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64035)
20:29:03.980 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:62224)
20:29:03.980 |  
20:29:03.980 | Error occurred prerendering page "/profile". Read more: https://nextjs.org/docs/messages/prerender-error
20:29:03.980 | ReferenceError: navigator is not defined
20:29:03.980 | at a (/vercel/path0/build/server/chunks/293.js:1:12801)
20:29:03.980 | at v (/vercel/path0/build/server/chunks/293.js:1:5186)
20:29:03.980 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:45321)
20:29:03.980 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.980 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.980 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:56390)
20:29:03.980 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
20:29:03.980 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
20:29:03.980 | at nO (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64035)
20:29:03.980 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:62224)

<!--EndFragment-->
</body>
</html>
```

**Comment by skks1212** [comment]
```
<html>
<body>
<!--StartFragment-->
ReferenceError: navigator is not defined
--
21:20:18.959 | at a (/vercel/path0/build/server/chunks/293.js:1:12801)
21:20:18.959 | at v (/vercel/path0/build/server/chunks/293.js:1:5186)
21:20:18.960 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:45321)
21:20:18.960 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
21:20:18.960 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
21:20:18.960 | at nP (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:56390)
21:20:18.965 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61978)
21:20:18.966 | at nj (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64628)
21:20:18.966 | at nO (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64035)
21:20:18.966 | at n$ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:62224)

<!--EndFragment-->
</body>
</html>
```

@rithviknishad is navigator being called server side? 🤔

**Comment by rithviknishad** [comment]
I think so. I've no clue why it's being called in SSR even after it being wrapped inside a useEffect

**Comment by rithviknishad** [comment]
weirdly, local builds are not reporting this

**Comment by rithviknishad** [comment]
@skks1212 @Ashesh3 fixed :)

**Review by skks1212 — APPROVED** [review]

**Pull request merged**